src/experimental/tools/VideoThumbnailLoader/thumbnail_loader.ts

```diff
@@ -0,0 +1,306 @@
+                                  { segmentLoader,
+                                    segmentParser }).pipe(
+        mergeMap((videoSourceBuffer) => {
+          const bufferCleaning$ = removeBufferAroundTime$(this._videoElement,
+                                                          videoSourceBuffer,
+                                                          time);
+          log.debug("VTL: Removing buffer around time.", time);
+
+          const segmentsLoading$ =
+            loadSegments(segments, this._requests, segmentLoader, contentInfos);
+
+          return observableMerge(bufferCleaning$.pipe(ignoreElements()),
+                                 segmentsLoading$
+          ).pipe(
+            mergeMap((arr) => {
+              return combineLatest(
+                arr.map(({ segment, data }) => {
+                  const start = segment.time / segment.timescale;
+                  const end = start + (segment.duration / segment.timescale);
+                  const inventoryInfos = objectAssign({ segment,
+                                                        start,
+                                                        end }, contentInfos);
+                  return pushData(inventoryInfos,
+                                  segmentParser,
+                                  data,
+                                  videoSourceBuffer)
+                    .pipe(tap(() => {
+                      this._requests.delete(segment.id);
+                      log.debug("VTL: Appended segment.", data);
+                    }));
+                })
+              );
+            }),
+            map(() => {
+              this._videoElement.currentTime = time;
+              return time;
+            })
+          );
+        }),
+        catchError((err: unknown) => {
+          let message = "Unknown error.";
+          if ((err as { message?: string; toString(): string }).message !== undefined ||
+            /* eslint-disable @typescript-eslint/no-unsafe-assignment */
+            /* eslint-disable @typescript-eslint/no-unsafe-member-access */
+            /* eslint-disable @typescript-eslint/no-unsafe-call */
+              typeof (err as { message?: string;
+                               toString(): string; }).toString === "function") {
+            message = (err as any).message ?? (err as any).toString();
+            /* eslint-enable @typescript-eslint/no-unsafe-assignment */
+            /* eslint-enable @typescript-eslint/no-unsafe-member-access */
+            /* eslint-enable @typescript-eslint/no-unsafe-call */
+          }
+          throw new VideoThumbnailLoaderError("LOADING_ERROR", message);
+        })
+      )
+    ).pipe(
+      take(1),
+      finalize(() => {
+        segments.forEach((segment) => {
+          if (this._nextTaskSegments === undefined ||
+              !this._nextTaskSegments.some(({ id }) => segment.id === id)) {
+            const request = this._requests.get(segment.id);
```

Comment:
Here also, I would guess that's more a teardown thing of the Observable created in `loadSegments`